src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = last = newNode;
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        checkPositionByElement(index);
+        if (isEmpty()) {
```

Comment:
isEmpty it the same that index == size so you can just call method add(T value) 
and [Don’t complicate if-else construction. ](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-complicate-if-else-construction-detailed-explanation)